### DISCUSSION TITLE
Is there a Figma for Mantine components?

### DISCUSSION THREAD

**Discussion by kungpaogao** [mantinedev/mantine#382] [Feedback]
Are the component designs based on some Figma file? And if so, could this be shared publicly?

**Comment by rtivital** [comment]
No, we currently cannot share our designs on Figma

**Reply by kungpaogao** [reply]
Are there any plans in the future? It would be great for designers and encouraging greater use of the components.

**Reply by rtivital** [reply]
I do not have much experience with Figma and I think it would be a huge job to conver everything to Figma components, I do not think that we will provide this feature soon.

**Reply by kungpaogao** [reply]
Oh, I see! I thought that the designs already existed in a Figma file somewhere. Are the designs of the components based on any mockups elsewhere, or do you just design as you implement it?

**Reply by rtivital** [reply]
All designs are built in browser only

**Reply by danols** [reply]
In The Zone has published a Mantine component library for Figma, https://www.figma.com/community/file/1067173247578645134. If you decide that you would like to develop a Mantine sponsored library, I would be more than happy to contribute. I'm a UI Designer w/expertise in Figma and managing design systems.

**Reply by Mike-Ro** [reply]
There is also a Figma component library (which we use): https://www.figma.com/community/file/1212329956432440320 (competition is good)!

**Comment by anthonyshew** [comment]
So glad to see that this question is being asked because I had the same one!

We are beginning to develop a product that will be using Mantine for our components. Our designer is beginning today on turning the components from the documentation into a Figma library. We fully plan on making it available for everyone if that is desired by you, @rtivital. We think we have a pretty good plan for mapping the Figma library to the components themselves so that anyone could use it effectively.

We really appreciate your work and would love to be able to help with what we can. Hoping that having a Figma library available will make Mantine more available for more people!

**Reply by rtivital** [reply]
Sure, feel free to share a Figma library, I'll add it to documentation

**Reply by anthonyshew** [reply]
Awesome! We will try to get as far as we can ourselves but may come to you for insights!

**Reply by LucasLemanowicz** [reply]
Very interested in this as well, please keep me in the loop @anthonyshew

**Reply by anthonyshew** [reply]
@LucasLemanowicz Absolutely will! She thinks she will be done in about a week from now so will keep you guys posted!

**Reply by anthonyshew** [reply]
Hey, everyone!

Here is the first iteration featuring four or five components just to get the ball rolling so that you guys can play with it a little bit! Any feedback is appreciated if you find any problems! (As mentioned, just a few components here thus far to make sure that she is on the right track.)

https://www.figma.com/community/file/1067173247578645134

She will be making a comment below in a minute or so here just to make herself known so you guys can tag her or ask questions!

P.S. (spam tagging everyone who has shown interest above for maximum visibility😊) @kungpaogao @LucasLemanowicz @albertbronsky @captchafree @edukure

**Reply by shewashley** [reply]
Hi all, I am Anthony's designer! Any feedback on what is created so far is more than welcome. :) The plan is to push out updates daily.

I should note that it will probably be a few weeks before the library is complete unlike previously mentioned. This is the first component library I've created, so I wasn't sure how long it would take.

I look forward to hearing what you all think!

**Reply by albertbronsky** [reply]
> Hi all, I am Anthony's designer! Any feedback on what is created so far is more than welcome. :) The plan is to push out updates daily.
> 
> 
> 
> I should note that it will probably be a few weeks before the library is complete unlike previously mentioned. This is the first component library I've created, so I wasn't sure how long it would take.
> 
> 
> 
> I look forward to hearing what you all think!

I've checked it and it looks awesome already! Thank you for your work!

Also, I encourage everyone to go and "heart" it on Figma to help promote the library so more people can find it.

**Reply by albertbronsky** [reply]
> Hey, everyone!
> 
> Here is the first iteration featuring four or five components just to get the ball rolling so that you guys can play with it a little bit! Any feedback is appreciated if you find any problems! (As mentioned, just a few components here thus far to make sure that she is on the right track.)
> 
> https://www.figma.com/community/file/1067173247578645134
> 
> She will be making a comment below in a minute or so here just to make herself known so you guys can tag her or ask questions!
> 
> P.S. (spam tagging everyone who has shown interest above for maximum visibility😊) @kungpaogao @LucasLemanowicz @albertbronsky @captchafree @edukure

@rtivital we should consider adding the link to this in Mantine's Docs/Getting started section since having design resources can be critical for some teams when evaluating Mantine as their library of choice. 
Here's [example in MUI docs](https://mui.com/getting-started/installation/#design-resources).

**Reply by LucasLemanowicz** [reply]
Looks great @shewashley! Can't wait to use it.

If you're looking for additional ideas, one thing that would be nice is to account for some of the optional props (eg: TextInput with no description is the one that comes to mind). Not sure if that's something that is supported out of the box with these components or if it requires adding more variants, but I imagine it would be super helpful for many people.

**Reply by shewashley** [reply]
@LucasLemanowicz All of the components are built with Auto Layout, meaning that if you remove anything inside of a component (like the Frame that holds that description, for example), the component should automatically resize!

**Reply by kungpaogao** [reply]
Looks great @shewashley! Glad that this post got some traction!

**Reply by dbrxnds** [reply]
Thank you very much @anthonyshew. And you too @shewashley, looks great already! Ofcourse feel free to reach out to the community if there is something we can help with.

**Reply by MrPeker** [reply]
Looks really great!

**Reply by Chrischuck** [reply]
Hi @anthonyshew and @shewashley, I appreciate your work on this, very helpful

**Reply by zbubric** [reply]
@anthonyshew and @shewashley, thank you for your work on this! This would boost (already magnificent) Mantine even further and, I'm quite sure, be a major 'selling point' for anybody considering using Mantine.

Trying to catch-up such extensive library in Figma is tremendous job but, I'm sure that it will become easier one you reach 'existing covered, just need to follow updates' point.

Just wondering, due to amount of work, did you consider some way to distribute workload to other contributors and acting as a 'coordinator' for overall progress? I'm sure that there are many people that would like to help with this one.

BR
Zoran

**Reply by underyx** [reply]
@anthonyshew @shewashley: If you have a way for us to collaborate, my company is also interested in contributing to the Figma library.

**Reply by anthonyshew** [reply]
@zbubric @underyx & any other spectators:

Honestly, Ashley threw this together for us so we could use it for our work and really didn't expect such a strong response from the community!

We will look into creating a more structured platform for this so others can contribute.

**Reply by anthonyshew** [reply]
We've created a repository where we're handling the figma library here!: https://github.com/anthonyshew/mantine-figma

**Reply by rcronin** [reply]
@anthonyshew is this no longer available?

**Comment by HamishMcDougall** [comment]
Will there be light and dark version of the components in the Figma file  ?

**Reply by danolsavsky** [reply]
We could add them together if you're interested. HMU.

**Comment by danolsavsky** [comment]
Do we have any updates? Just in case you missed it, I volunteered to contribute to an official Mantine Figma library. The old library (authored by In The Zone) is no longer online at the moment. Please let me know if we could connect. Thanks!

**Reply by rtivital** [reply]
There is no official Mantine Figma library

**Comment by AlexKorzh** [comment]
Hi @rtivital! We are going to use Mantine for building our React application.
As of now, I haven't found a suitable Figma UI kit, so my question is how to proceed if there is no suitable Figma kit?
What you will suggest, how to act our designer in that case?

**Reply by rtivital** [reply]
You can find available figma libraries on getting started page – https://mantine.dev/getting-started/#mantine-for-figma

**Reply by entroid** [reply]
"Design is not a part of the development process " How shortsighted and detached from modern workflows.  So I'm courious, you guys just start randomly throwing out components on a page and hope something comes out of that?
